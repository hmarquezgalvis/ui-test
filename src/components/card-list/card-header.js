import React, { useContext } from 'react'
import styled from 'styled-components';
import { RoulingContext } from '../contexts/rouling.context';

import { DropDown } from '../common/dropdown/dropdown';

export const CardListHeader = ({ title, showModeSelector = true }) => {
  const {
    layout,
    setLayout,
    RESULT_LAYOUTS,
  } = useContext(RoulingContext);
  return (
    <HeaderContainer>
      <h2>{title}</h2>
      {showModeSelector && (
        <DropDown
          options={RESULT_LAYOUTS}
          value={layout}
          onSelected={(layout) => setLayout(layout)}
        />
      )}
    </HeaderContainer>
  )
};

const HeaderContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 24px;
  @media (min-width: 768px) {
    margin-bottom: 28px;
  }
  @media (min-width: 1100px) {
    margin-bottom: 36px;
  }
  h2 {
    margin: 0;
    color: #464646;
    font-family: 'Lato';
    font-style: normal;
    font-weight: 300;
    font-size: 24px;
    line-height: 29px;
    @media (min-width: 1100px) {
      font-size: 45px;
      line-height: 54px;
    }
  }
`;