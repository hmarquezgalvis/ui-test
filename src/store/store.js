import { configureStore } from '@reduxjs/toolkit';

import roulingsReducer from '../reducers/rouling.reducer';
import myVotesReducer from '../reducers/my-votes.reducer';

export default configureStore({
  reducer: {
    roulings: roulingsReducer,
    profile: myVotesReducer,
  },
});