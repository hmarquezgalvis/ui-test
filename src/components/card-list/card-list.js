import React, { useContext } from 'react';
import styled from 'styled-components';
import { useMediaQuery } from 'react-responsive';

import { RoulingContext } from '../contexts/rouling.context';
import { RESULT_LAYOUT_GRID } from '../hooks/use-rouling-filter';
import { CardListHeader } from './card-header';
import { CARD_TYPE } from '../../types/card';
import { Card } from '../card';

export const CardList = () => {
  const { 
    results, 
    updateVote,
    layout, 
  } = useContext(RoulingContext);
  const isSmallScreen = useMediaQuery({ query: '(max-width: 767px)' });
  const showAsGrid = layout === RESULT_LAYOUT_GRID || isSmallScreen;
  const cardType = showAsGrid ? CARD_TYPE.GRID_ITEM : CARD_TYPE.LIST_ITEM;
  
  return (
    <>
      <CardListHeader 
        title="Previous Rulings"
        showModeSelector={!isSmallScreen}
      />
      <CardListContainer 
        showAsGrid={showAsGrid}
      >
        {results && results.map((item, index) => (
          <Card key={item.guid} item={item} updateVote={updateVote} type={cardType} />
        ))}
      </CardListContainer>
    </>
  );
};

const CardListContainer = styled.div`
  display: flex;
  flex-wrap: nowrap;
  gap: 12px;
  overflow: scroll hidden;
  scroll-snap-type: x mandatory;

  & > div {
    scroll-snap-align: start;
  }

  @media all and (min-width: 768px) {
    display: grid;
    grid-template-columns: ${(props) => props.showAsGrid ? '1fr 1fr' : '1fr'};
    gap: 21px;
    overflow: hidden;
  }

  @media all and (min-width: 1100px) {
    grid-template-columns: ${(props) => props.showAsGrid ? '1fr 1fr 1fr' : '1fr'};
    gap: 28px;
  }
`;