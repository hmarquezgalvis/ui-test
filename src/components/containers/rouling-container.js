import React from 'react'
import { CardList } from '../card-list/card-list';
import { RoulingProvider } from '../contexts/rouling.context';

export const RoulingContainer = () => {
  return (
    <RoulingProvider>
      <CardList />
    </RoulingProvider>
  );
};