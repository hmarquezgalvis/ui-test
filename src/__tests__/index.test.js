import { render, screen } from '@testing-library/react';
import { CardList } from '../components/card-list/card-list';
import { RoulingContainer } from '../components/containers/rouling-container';

import { Provider } from 'react-redux';
import store from '../store/store';

describe('tests of Main', () => {
  
  it('should render and save the snapshot correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <RoulingContainer><CardList /></RoulingContainer>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/Previous Rulings/i)).toBeInTheDocument();
  });

});