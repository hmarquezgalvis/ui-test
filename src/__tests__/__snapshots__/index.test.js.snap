// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests of Main should render and save the snapshot correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-eCYdqJ ectmOT"
  >
    <h2>
      Previous Rulings
    </h2>
    <div
      class="sc-bczRLJ bGOKaH"
    >
      <div
        class="sc-gsnTZi kSpeQC current"
      >
        List 
        <span
          class="chevron"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-fLlhyt bYkOFR"
  />
</DocumentFragment>
`;
